src/nvim/quickfix.c

```diff
@@ -892,19 +965,20 @@ static int
 qf_init_ext(
     qf_info_T *qi,
     int qf_idx,
-    char_u *efile,
+    const char_u *efile,
     buf_T *buf,
     typval_T *tv,
     char_u *errorformat,
-    int newlist,                            // TRUE: start a new error list
+    bool newlist,                           // true: start a new error list
     linenr_T lnumfirst,                     // first line number to use
     linenr_T lnumlast,                      // last line number to use
     char_u *qf_title,
     char_u *enc
 )
+  FUNC_ATTR_NONNULL_ARG(1)
 {
-  qfstate_T state;
-  qffields_T fields;
+  qfstate_T state = { 0 };
+  qffields_T fields = { 0 };
```

Comment:
No `Wmissing-field-initializer` warning this time unlike https://github.com/neovim/neovim/pull/9994.